airflow/providers/salesforce/operators/tableau_refresh_workbook.py

```diff
@@ -0,0 +1,74 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from airflow import AirflowException
+from airflow.models import BaseOperator
+from airflow.providers.salesforce.hooks.tableau import TableauHook
+from airflow.utils.decorators import apply_defaults
+
+
+class TableauRefreshWorkbookOperator(BaseOperator):
+    """
+    Refreshes a Tableau Workbook/Extract
+
+    :param workbook_name: The name of the workbook to refresh.
+    :type workbook_name: str
+    :param site_id: The id of the site where the workbook belongs to.
+    :type site_id: str
+    :param tableau_conn_id: The Tableau Connection id containing the credentials
+        to authenticate to the Tableau Server.
+    :type tableau_conn_id: str
+    """
+
+    @apply_defaults
+    def __init__(self,
+                 workbook_name,
+                 *args,
+                 site_id='',
+                 tableau_conn_id='tableau_default',
+                 **kwargs):
+        super().__init__(*args, **kwargs)
+        self.workbook_name = workbook_name
+        self.site_id = site_id
+        self.tableau_conn_id = tableau_conn_id
+
+    def execute(self, context):
+        """
+        Executes the Tableau Extract Refresh and pushes the job id to xcom.
+
+        :param context: The task context during execution.
+        :type context: dict
+        :return: the id of the job that executes the extract refresh
+        :rtype: str
+        """
+        with TableauHook(self.site_id, self.tableau_conn_id) as tableau_hook:
+            workbook = self._get_workbook_by_name(tableau_hook)
+
+            return self._refresh_workbook(tableau_hook, workbook.id)
+
+    def _get_workbook_by_name(self, tableau_hook):
+        for workbook in tableau_hook.get_all(resource_name='workbooks'):
+            if workbook.name == self.workbook_name:
+                self.log.info('Found matching workbook with id %s', workbook.id)
+                return workbook
+
+        raise AirflowException(f'Workbook {self.workbook_name} not found!')
+
+    def _refresh_workbook(self, tableau_hook, workbook_id):
+        job = tableau_hook.server.workbooks.refresh(workbook_id)
```

Comment:
Does this support refreshing individual workbook report or the whole workbook ?
Say I have:
`https://url/#/site/site_name/views/Calls/Department`
`https://url/#/site/site_name/views/Calls/Agents`
if I pass `Calls` would it refresh both `Department` & `Agents` ?
What if I pass `Calls/Agents` ?

Reply:
I am gonna check that.

Reply:
> Refreshes the specified workbook.
> 
> This method refreshes the specified workbook, with no need to associate the workbook refresh with a scheduled task. This method is the equivalent of selecting a workbook using the Tableau Server UI, and then selecting Refresh Extracts from the menu (also known as a "manual refresh").
> 

from https://help.tableau.com/current/api/rest_api/en-us/REST/rest_api_ref_workbooksviews.htm#update_workbook_now

Reply:
> if I pass Calls would it refresh both Department & Agents ?

Yes.

> What if I pass Calls/Agents ?

I think it fails, because that's not a Workbook - it is a View in a Workbook, right?!

Reply:
You actually pass an ID of a workbook not the name. I am iterating through workbooks by names and I am returning the workbook id of the workbook that matches.